pallets/rmrk-core/src/lib.rs

```diff
@@ -230,54 +228,32 @@ pub mod pallet {
 			origin: OriginFor<T>,
 			owner: T::AccountId,
 			collection_id: CollectionId,
-			recipient: T::AccountId,
-			royalty: Permill,
+			recipient: Option<T::AccountId>,
+			royalty: Option<Permill>,
 			metadata: BoundedVec<u8, T::StringLimit>,
 		) -> DispatchResult {
 			let sender = match T::ProtocolOrigin::try_origin(origin) {
 				Ok(_) => None,
 				Err(origin) => Some(ensure_signed(origin)?),
 			};
 
-			let collection =
-				Self::collections(collection_id).ok_or(Error::<T>::CollectionUnknown)?;
-
-			let nfts_minted = NFTs::<T>::iter_prefix_values(collection_id).count();
-			let max: u32 = collection.max.try_into().unwrap();
-
-			ensure!(
-				// Probably a better way to do "max == 0"
-				nfts_minted < max.try_into().unwrap() || max == max - max,
-				Error::<T>::CollectionFullOrLocked
-			);
-
-			let nft_id: NftId = Self::get_next_nft_id(collection_id)?;
-
-			// let metadata_bounded = Self::to_bounded_string(metadata)?;
-			// if let Some(r) = royalty {
-			// 	ensure!(r < 1000, Error::<T>::NotInRange);
-			// }
-
-			// pallet_uniques::Pallet::<T>::do_mint(
-			// 	collection_id,
-			// 	nft_id,
-			// 	sender.clone().unwrap_or_default(),
-			// 	|_details| Ok(()),
-			// )?;
-
 			let rootowner = owner.clone();
-			let owner_as_maybe_account = AccountIdOrCollectionNftTuple::AccountId(owner.clone());
 
-			let nft = InstanceInfo {
-				owner: owner_as_maybe_account,
-				rootowner,
+			let (collection_id, nft_id) = <Self as Nft<T::AccountId, StringLimitOf<T>>>::mint_nft(
+				sender.clone().unwrap_or_default(),
```

Comment:
When `sender == None`? If it means the origin is some admin (ProtocolOrigin), I suggest to use pallet's account id instead of the default account id (0x0), which is going to be deprecated soon due to [some security concern](https://github.com/paritytech/substrate/pull/10403).

Reply:
Example of a pallet owned account_id:

https://github.com/paritytech/substrate/blob/d18705ee7b5b654d3947558fea6dbd0701ac3ecc/frame/society/src/lib.rs#L1733-L1735

Reply:
Will create separate issue